server/aesdsocket.c

```diff
@@ -37,9 +39,25 @@ int create_socket();
 void set_socket_options(int sockfd);
 void bind_socket(int sockfd);
 void listen_for_connections(int sockfd);
-void handle_client(int client_fd);
+void *handle_client(void *arg);
 void accept_clients(int sockfd);
 void daemonize();
+void add_thread(pthread_t tid);
+void remove_thread(pthread_t tid);
+void join_completed_threads();
+void *append_timestamp(void *arg);
+
+typedef struct ThreadNode {
+    pthread_t tid;
+    struct ThreadNode *next;
+} ThreadNode;
+
+ThreadNode *thread_list = NULL; 
+
+volatile bool sig_exit = false;
+int custom_socket_fd = -1;
+pthread_mutex_t list_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
@Suhas-Reddy-S I think you could simplify the design and remove the need for this mutex if you only manipulated the list in the main function which started your threads and not in the threads themselves.